From 1fb7e0dfb19b9cd269a3940f60a100702940f5c1 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Sun, 30 May 2021 21:46:34 +0300
Subject: [PATCH 247/302] media: i2c: imx296: Expose crop bounds and defaults

The crop bounds and defaults are needed by libcamera. Expose them.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 drivers/media/i2c/imx296.c | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/drivers/media/i2c/imx296.c b/drivers/media/i2c/imx296.c
index db5a5c293feb..5da481d09901 100644
--- a/drivers/media/i2c/imx296.c
+++ b/drivers/media/i2c/imx296.c
@@ -732,10 +732,23 @@ static int imx296_get_selection(struct v4l2_subdev *sd,
 {
 	struct imx296 *imx = to_imx296(sd);
 
-	if (sel->target != V4L2_SEL_TGT_CROP)
+	switch (sel->target) {
+	case V4L2_SEL_TGT_CROP:
+		sel->r = *imx296_get_pad_crop(imx, cfg, sel->pad, sel->which);
+		break;
+
+	case V4L2_SEL_TGT_CROP_DEFAULT:
+	case V4L2_SEL_TGT_CROP_BOUNDS:
+	case V4L2_SEL_TGT_NATIVE_SIZE:
+		sel->r.left = 0;
+		sel->r.top = 0;
+		sel->r.width = IMX296_PIXEL_ARRAY_WIDTH;
+		sel->r.height = IMX296_PIXEL_ARRAY_HEIGHT;
+		break;
+
+	default:
 		return -EINVAL;
-
-	sel->r = *imx296_get_pad_crop(imx, cfg, sel->pad, sel->which);
+	}
 
 	return 0;
 }
-- 
Regards,

Laurent Pinchart

